2-structured-data/src/DataModel/Datastore.php

```diff
@@ -151,106 +104,37 @@ public function update($book)
         $this->verifyBook($book);
 
         if (!isset($book['id'])) {
-            throw new InvalidArgumentException('Book must have an "id" attribute');
+            throw new \InvalidArgumentException('Book must have an "id" attribute');
         }
 
-        $key = $this->createKey($book['id']);
-        $properties = $this->bookToProperties($book);
-
-        $entity = new \Google_Service_Datastore_Entity([
-            'key' => $key,
-            'properties' => $properties
-        ]);
-
-        // Use "NON_TRANSACTIONAL" for simplicity (as we're only making one call)
-        $request = new \Google_Service_Datastore_CommitRequest([
-            'mode' => 'NON_TRANSACTIONAL',
-            'mutations' => [
-                [
-                    'update' => $entity
-                ]
-            ]
-        ]);
-
-        $response = $this->datastore->projects->commit($this->datasetId, $request);
+        $transaction = $this->datastore->transaction();
+        $key = $this->datastore->key('Book', $book['id']);
+        $task = $transaction->lookup($key);
+        unset($book['id']);
```

Comment:
The "data model" interface expects an array with the "id" property set. But if you save the model with that property, the Datastore API throws an error.